src/debug/debug-pal/unix/diagnosticsipc.cpp

```diff
@@ -0,0 +1,107 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#include <pal.h>
+#include <pal_assert.h>
+#include <new>
+#include <unistd.h>
+#include <fcntl.h>
+#include <sys/types.h>
+#include <sys/socket.h>
+#include <sys/un.h>
+#include "diagnosticsipc.h"
+
+IpcStream::DiagnosticsIpc::DiagnosticsIpc(const char *const pIpcName, const uint32_t pid) :
+    _pServerAddress(new (std::nothrow) sockaddr_un)
+{
+    _ASSERTE(_pServerAddress != nullptr);
+    _ASSERTE(pIpcName != nullptr);
+
+    _serverSocket = ::socket(PF_UNIX, SOCK_STREAM, 0);
```

Comment:
Do we have to perform this initialization it in the constructor?
There are no way you can bubble this error up without throwing exception.